### PR TITLE
Added support for viewing markdown documents.

### PR DESCRIPTION
**Pull request by sarmadka**
* Added MarkdownTranslator utility class for converting from markdown to html.
* Added DocView widget for rendering full documents from html or markdown sources.
* Added RegExp class and renamed Label to Text.
* Fixed some bugs with regular expressions.
* Allow class based styling on the component itself rather than only on its children.
* Enabled differentiating between immediate children and non immediate children when defining styles.
* Enabled styling children by tag name.
* Fixed a bug with RoutingSwitcher causing unnecessary instantiation of components when route events result in staying on the same component.
* Added some missing Arabic localizations and moved utility files into Utils folder.
* Updated the routing example to use DocView for displaying MD formatted text.

### CODE REVIEW

File: WebPlatform/Utils/string_operations.alusus [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        return matchRegex(str, regex, lastIndex);
+    }
+
+    function matchRegex (str: ptr[array[Char]], regexId: ArchInt, lastIndex: ref[Int]): Array[String] {
+        return parseRegexMatchResult(_matchRegex(str, 0, regexId, lastIndex));
+    }
+
+    function matchRegex (str: ptr[array[Char]], regexId: ArchInt): Array[String] {
+        def lastIndex: Int;
+        return matchRegex(str, regexId, lastIndex);
```
Comment (SlemanAlrstum231) [inline WebPlatform/Utils/string_operations.alusus:17]: سؤال اخي الكريم هنا يمرر بشكل افتراضي int بالعنوان

File: WebPlatform/Utils/string_operations.alusus [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        return parseRegexMatchResult(_matchRegex(str, regex, 0, lastIndex));
+    }
+
+    function matchRegex (str: ptr[array[Char]], regex: ptr[array[Char]]): Array[String] {
+        def lastIndex: Int;
+        return matchRegex(str, regex, lastIndex);
+    }
+
+    function matchRegex (str: ptr[array[Char]], regexId: ArchInt, lastIndex: ref[Int]): Array[String] {
+        return parseRegexMatchResult(_matchRegex(str, 0, regexId, lastIndex));
```
Comment (SlemanAlrstum231) [inline WebPlatform/Utils/string_operations.alusus:12]: واني تعرفي هذا التابع اخي الكريم _matchRegex

Comment (sarmadka) [reply WebPlatform/Utils/string_operations.alusus:12]: تعريفه في ملف browser_api

File: WebPlatform/Utils/string_operations.alusus (hunk 1 above)
Comment (sarmadka) [reply WebPlatform/Utils/string_operations.alusus:17]: لم أفهم السؤال، ماذا تقصد بالعنوان؟

**Review by SlemanAlrstum231 — DISMISSED** [review]

**Pull request merged**